### PR TITLE
Fix missing operator-generic in upstream k8s tests

### PR DESCRIPTION
**Pull request by aanm**
This PR fixes some bugs introduced by https://github.com/cilium/cilium/pull/12029

### CODE REVIEW

**Comment by aanm** [comment]
test-upstream-k8s

**Comment by aanm** [comment]
test-upstream-k8s

File: test/provision/compile.sh [hunk 1]
```diff
@@ -45,13 +48,28 @@ then
 
       if [[ "${CILIUM_OPERATOR_IMAGE}" == "" ]]; then
         echo "building cilium-operator image..."
-        make LOCKDEBUG=1 docker-operator-image
-        echo "tagging cilium-operator image..."
+        make LOCKDEBUG=1 docker-operator-image docker-operator-aws-image docker-operator-azure-image docker-operator-generic-image
```
Comment (joestringer) [inline test/provision/compile.sh]: We should probably reference https://github.com/cilium/cilium/issues/12035 here as well

(The equivalent change for the regular cilium build is in the makefile)

**Review by joestringer — APPROVED** [review]
Let's see if it passes this time.

**Review by tgraf — APPROVED** [review]

**Review by rolinh — APPROVED** [review]

**Comment by aanm** [comment]
test-upstream-k8s

**Comment by aanm** [comment]
test-me-please

**Pull request merged**